core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -149,14 +150,15 @@ static CloseableIterable<FileScanTask> planFiles(StaticTableScan scan) {
     Snapshot snapshot = table.snapshot(scan.snapshot().snapshotId());
     boolean caseSensitive = scan.isCaseSensitive();
 
-    // use an inclusive projection to remove the partition name prefix and filter out any non-partition expressions
-    Expression partitionFilter = Projections
-        .inclusive(transformSpec(scan.schema(), table.spec()), caseSensitive)
-        .project(scan.filter());
+    LoadingCache<Integer, ManifestEvaluator> evalCache = Caffeine.newBuilder().build(specId -> {
```

Comment:
Not sure we need the full LoadingCache here, but I'm ok with it if you like, we could probably just proactively build the full set of evaluators for all specs in the metadata. I probably should have suggested this before on the other PR as well. Since we know we will need every single evaluator

Reply:
Discussed, at this point we keep all partition specs so this will very slightly save a cycle if we have some spec without any manifests.